connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/lifecycle/InboundConnectorManager.java

```diff
@@ -167,7 +167,19 @@ private void activateConnector(InboundConnectorDefinitionImpl newConnector) {
       if (executable instanceof PollingConnectorExecutable) {
         Long processDefinitionKey = newConnector.processDefinitionKey();
         Supplier<List<ProcessInstance>> processInstancesSupplier =
-            () -> processDefinitionSearch.fetchProcessInstancesWithVariables(processDefinitionKey);
+            () ->
+                processDefinitionSearch
+                    .fetchProcessInstancesByDefinitionKey(processDefinitionKey)
+                    .stream()
+                    .map(
+                        instance ->
+                            new ProcessInstance(
+                                instance.getKey(),
+                                () ->
+                                    processDefinitionSearch.fetchVariablesByProcessInstanceKey(
+                                        instance.getKey())))
+                    .collect(Collectors.toList());
```

Comment:
We should `diff` against finished instances and new/running instances first without fetching the variables. Only fetch the variables when you need them (creating the polling request).